pkg/client/client.go

```diff
@@ -107,17 +107,42 @@ func NewDefaultClientWithTimeout(timeout time.Duration) (*Client, error) {
 // If host is nil then use SockPath provided by CILIUM_SOCK
 // or the cilium default SockPath
 func NewClient(host string) (*Client, error) {
-	clientTrans, err := NewRuntime(host, "")
+	clientTrans, err := NewRuntime(WithHost(host))
 	return &Client{*clientapi.New(clientTrans, strfmt.Default)}, err
 }
 
-func NewRuntime(host, basePath string) (*runtime_client.Runtime, error) {
+type runtimeOptions struct {
+	host     string
+	basePath string
+}
+
+func WithHost(host string) func(options *runtimeOptions) {
+	return func(options *runtimeOptions) {
+		options.host = host
+	}
+}
+
+func WithBasePath(basePath string) func(options *runtimeOptions) {
+	return func(options *runtimeOptions) {
+		options.basePath = basePath
+	}
+}
+
+func NewRuntime(opts ...func(options *runtimeOptions)) (*runtime_client.Runtime, error) {
+	r := runtimeOptions{}
+	for _, opt := range opts {
+		opt(&r)
+	}
+
+	host := r.host
```

Comment:
I had it like this originally, but that breaks the current call sites. The current callers may call `WithHost("")` (e.g. in articular [this one here](https://github.com/cilium/cilium/blob/a5bc6acb91a0f5a1ce886da41de777083e15f493/cilium-dbg/cmd/map_event_list.go#L44)). I would have to move this `if host == ""` check up to the callers to ensure they are not overwriting the `DefaultSockPath()` with an empty string.

So the trade off here is making things more complicated for the callers or have `WithHost("")` treated specially (which the current version of the commit does). I have no particular preference, if you feel moving things to the caller is the better option I'm happy to change.

Reply:
argh :-/ moving to the caller isn't that nice either.

an additional option would be to prevent the option-functions to accept an empty string where it doesn't make any sense (silently dropping in favor of the meaningful default)

Reply:
If we treat `WithHost("")` specially, I prefer the current "centralized" solution over moving the special cases into the option-functions. In particular since the remaining code also does some tweaking to the `host` variable (e.g. replacing "tcp" for "http" etc), and so having all the logic in one place seems cleanest.

Reply:
agree